x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/sp/CloudServiceProvider.java

```diff
@@ -27,13 +29,16 @@
     private final boolean signAuthnRequests;
     private final boolean signLogoutRequests;
 
-    public CloudServiceProvider(String entityId, URL assertionConsumerService, Set<String> allowedNameIdFormats,
-                                ReadableDuration authnExpiry, ServiceProviderPrivileges privileges, AttributeNames attributeNames,
+    public CloudServiceProvider(String entityId, String name, boolean enabled, URL assertionConsumerService,
+                                Set<String> allowedNameIdFormats, ReadableDuration authnExpiry,
+                                ServiceProviderPrivileges privileges, AttributeNames attributeNames,
                                 Set<X509Credential> spSigningCredentials, boolean signAuthnRequests, boolean signLogoutRequests) {
         if (Strings.isNullOrEmpty(entityId)) {
             throw new IllegalArgumentException("Service Provider Entity ID cannot be null or empty");
         }
         this.entityId = entityId;
+        this.name = name;
```

Comment:
Do we need a name ?

Reply:
No, but we store it in the doc, so I bubbled it up.
We could rip it out entirely instead.